bug(prefer-timer-args): unbound method breaks `this` when calling method

`e18e/prefer-timer-args` breaks when a function inside a `setInterval`/`setTimeout` attempts to access the class instance using `this.`.

It also collides with [@typescript-eslint/unbound-method](https://typescript-eslint.io/rules/unbound-method), which attempts to prevent bugs like this.
```
  11:21  error  A method that is not declared with `this: void` may cause unintentional scoping of `this` when separated from its object.
Consider using an arrow function or explicitly `.bind()`ing the method to avoid calling the method with an unintended `this` value.
If a function does not access `this`, it can be annotated with `this: void`  @typescript-eslint/unbound-method
```

This issue differs from #46 in one key way:
Instead of calling `setTimeout(this.log, 1);` inside the `Bug` class, the `log` method attempts to access `this.MESSAGE` when I attempt to run `setTimeout(bug.log, 1);`

## Before: Working

<img width="666" height="503" alt="Image" src="https://github.com/user-attachments/assets/21d7af7c-4eb4-4d79-9847-06ff40cdc128" />

## After: Broken

<img width="1168" height="697" alt="Image" src="https://github.com/user-attachments/assets/0bc3b079-7ca9-4ac5-a938-27c4b6240203" />

--- a/src/rules/prefer-timer-args.test.ts
+++ b/src/rules/prefer-timer-args.test.ts
@@ -76,14 +76,17 @@ ruleTester.run('prefer-timer-args', preferTimerArgs, {
     'setTimeout(() => fn())',
     'notSetTimeout(() => doSomething(a), 100)',
 
+    // Method calls should not be transformed (could lose `this` binding)
+    'setTimeout(() => obj.method(arg), 100)',
+    'setTimeout(() => arr.filter(fn).map(transform), 100)',
+
     // References to `this` should not be transformed (would lose binding)
     'setTimeout(() => this.foo(), 100)',
     'setTimeout(() => this.handler.process(data), 500)',
     'setTimeout(() => fn(this.value), 100)',
     'setTimeout(() => fn(a, this.b, c), 100)',
     'setInterval(() => this.tick(), 1000)',
-    'window.setTimeout(() => this.render(), 16)',
-    'setTimeout(this.method.bind(null, arg), 100)'
+    'window.setTimeout(() => this.render(), 16)'
   ],
 
   invalid: [
@@ -124,17 +127,6 @@ ruleTester.run('prefer-timer-args', preferTimerArgs, {
     {
       code: 'setTimeout(() => process(a, b, c), 0)',
       output: 'setTimeout(process, 0, a, b, c)',
-      errors: [
-        {
-          messageId: 'preferArgs'
-        }
-      ]
-    },
-
-    // Method calls
-    {
-      code: 'setTimeout(() => obj.method(arg), 100)',
-      output: 'setTimeout(obj.method, 100, arg)',
       errors: [
         {
           messageId: 'preferArgs'
@@ -191,17 +183,6 @@ setTimeout(fn2, 200, b, c);`,
       ]
     },
 
-    // Chained method calls
-    {
-      code: 'setTimeout(() => arr.filter(fn).map(transform), 100)',
-      output: 'setTimeout(arr.filter(fn).map, 100, transform)',
-      errors: [
-        {
-          messageId: 'preferArgs'
-        }
-      ]
-    },
-
     // bind() with null context
     {
       code: 'setTimeout(fn.bind(null, a, b), 100)',
@@ -250,6 +231,17 @@ setTimeout(fn2, 200, b, c);`,
     {
       code: 'setTimeout(obj.method.bind(null, arg), 100)',
       output: 'setTimeout(obj.method, 100, arg)',
+      errors: [
+        {
+          messageId: 'preferArgs'
+        }
+      ]
+    },
+
+    // bind() on this.method with null context
+    {
+      code: 'setTimeout(this.method.bind(null, arg), 100)',
+      output: 'setTimeout(this.method, 100, arg)',
       errors: [
         {
           messageId: 'preferArgs'

--- a/src/rules/prefer-timer-args.ts
+++ b/src/rules/prefer-timer-args.ts
@@ -1,10 +1,5 @@
 import type {Rule} from 'eslint';
-import type {
-  CallExpression,
-  ArrowFunctionExpression,
-  Expression,
-  SpreadElement
-} from 'estree';
+import type {CallExpression, Expression, SpreadElement} from 'estree';
 
 function isNullOrUndefined(node: Expression): boolean {
   if (node.type === 'Literal' && node.value === null) {
@@ -122,26 +117,10 @@ export const preferTimerArgs: Rule.RuleModule = {
   },
   create(context) {
     const sourceCode = context.sourceCode;
-    let sawThis = false;
 
     return {
-      ThisExpression() {
-        sawThis = true;
-      },
-
       CallExpression(node: CallExpression) {
-        if (isTimerCall(node)) {
-          sawThis = false;
-        }
-      },
-
-      'CallExpression:exit'(node: CallExpression) {
         if (!isTimerCall(node)) {
-          return;
-        }
-
-        // Skip if the callback references `this` - transformation would lose binding
-        if (sawThis) {
           return;
         }
 
@@ -161,25 +140,27 @@ export const preferTimerArgs: Rule.RuleModule = {
 
         // simple arrow functions, e.g. () => fn(args)
         if (firstArg.type === 'ArrowFunctionExpression') {
-          const arrowFn = firstArg as ArrowFunctionExpression;
-
           // skip if it is a block body
-          if (arrowFn.body.type === 'BlockStatement') {
+          if (firstArg.body.type === 'BlockStatement') {
             return;
           }
 
           // skip if it has parameters
-          if (arrowFn.params.length > 0) {
+          if (firstArg.params.length > 0) {
             return;
           }
 
-          if (arrowFn.body.type !== 'CallExpression') {
+          if (firstArg.body.type !== 'CallExpression') {
             return;
           }
 
-          const callExpression = arrowFn.body;
+          const callExpression = firstArg.body;
           const callee = callExpression.callee;
           const callArgs = callExpression.arguments;
+
+          if (callee.type === 'MemberExpression') {
+            return;
+          }
 
           if (!callArgs.every(isSafeArgument)) {
             return;

